src/components/views/right_panel/RoomSummaryCard.tsx

```diff
@@ -200,14 +201,23 @@ const RoomSummaryCard: React.FC<IProps> = ({ room, onClose }) => {
 
     const isRoomEncrypted = useIsEncrypted(cli, room);
 
+    const [e2eStatus, setE2eStatus] = useState<string>();
+    useEffect(() => {
+        if (isRoomEncrypted) {
+            shieldStatusForRoom(cli, room).then(e => setE2eStatus(e));
+        }
+    });
```

Comment:
Is there no js-sdk events we could be watching here? This doesn't look like it'll update if the shield status changes in the room whilst the panel is open which will likely make it mismatch the other shields you see around the UI.

`useEventEmitter` is quite useful for watching events

Reply:
It should be possible base it on events, see e.g. the callers of `_updateE2EStatus` in `RoomView.js`. Events like `deviceVerificationChanged`, `userTrustStatusChanged`, `crossSigning.keysChanged`, as well as timeline events to enabled encryption are observed to only updated shields when needed.

Reply:
Actually. Is it in the state of RoomView? If so we can pick it up using `useContext(RoomContext)` which contains the entire RoomView state

Reply:
I'm failing to understand this conversation - what are the chances I can pawn this onto someone else while I go figure out what a hook is?

Using them in anger is not working.